### PR TITLE
Adding ManualCameraControl.cs and test scene

### PR DESCRIPTION
**Pull request by jochao**
ManualCameraControl.cs is a script that you can add to the main camera in order to control the camera with keyboard and mouse in unity preview mode.  (This script is pulled from galaxy explorer)

ManualCameraControl.unity is a test scene that shows how to use the script. When play mode is entered, you can move with keyboard and look around using ctrl + mouse.

### CODE REVIEW

**Comment by jwittner** [comment]
This looks great! We've been using something similar. I haven't tested it's functionality out, but was wondering how closely the controls match those in the emulator? For the toolkit I think something targeting parity with the emulator controls would be the way to go as it would alleviate mental remapping when going between the editor and emulator.

File: Assets/HoloToolkit/Utilities/Scripts/ManualCameraControl.cs [hunk 1]
```diff
@@ -0,0 +1,313 @@
...
+                UnityEngine.Cursor.visible = true;
+            }
+
+#if UNITY_EDITOR
+            UnityEditor.EditorGUIUtility.SetWantsMouseJumping(wantsJumping ? 1 : 0);
+#endif
+        }
+    }
+
+#if UNITY_EDITOR
```
Comment (jwittner) [inline Assets/HoloToolkit/Utilities/Scripts/ManualCameraControl.cs]: I wonder about moving this functionality out to it's own utility class. I think it'd make this class a little cleaner and more concise, and the the remote desktop check seems generally useful!

File: Assets/HoloToolkit/Utilities/Scripts/ManualCameraControl.cs [hunk 2]
```diff
@@ -0,0 +1,313 @@
...
+    private bool IsRunningUnderRemoteDesktop()
+    {
```
Comment (jwittner) [inline Assets/HoloToolkit/Utilities/Scripts/ManualCameraControl.cs]: I wonder about moving this functionality out to it's own utility class. I think it'd make this class a little cleaner and more concise, and the the remote desktop check seems generally useful!

**Comment by NeerajW** [comment]
Emulator seems like something we should investigate but I imagine people customizing these keys to fit their convenience.

File: Assets/HoloToolkit/Utilities/Scripts/ManualCameraControl.cs [hunk 3]
```diff
@@ -0,0 +1,313 @@
+﻿// Copyright Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+using UnityEngine;
+
+/// <summary>
+/// Class for manually controlling the camera when HuP isn't running. Attach to same CameraRig object as the StereoCamera.cs script.
```
Comment (NeerajW) [inline Assets/HoloToolkit/Utilities/Scripts/ManualCameraControl.cs]: What is HuP?

Comment (NeerajW) [inline Assets/HoloToolkit/Utilities/Scripts/ManualCameraControl.cs]: Perhaps clarifying in the summary comment that its the MainCamera object.

File: Assets/HoloToolkit/Utilities/Scripts/ManualCameraControl.cs [hunk 4]
```diff
@@ -0,0 +1,313 @@
...
+        Left,
+        Right,
+        Middle,
+        Control,
+        Shift,
+        Focused,
+        None
+    }
+
+    public float ExtraMouseSensitivityScale = 3.0f;
```
Comment (NeerajW) [inline Assets/HoloToolkit/Utilities/Scripts/ManualCameraControl.cs]: Do we need comments around the enum and public floats? ToolTips?

File: Assets/HoloToolkit/Utilities/Scripts/ManualCameraControl.cs [hunk 5]
```diff
@@ -0,0 +1,313 @@
...
+
+    private bool isGamepadLookEnabled = true;
+    private bool isFlyKeypressEnabled = true;
+    private Vector3 lastMousePosition = Vector3.zero;
+    private Vector3 lastTrackerToUnityTranslation = Vector3.zero;
+    private Quaternion lastTrackerToUnityRotation = Quaternion.identity;
+    private bool appHasFocus = true;
+    private bool wasLooking = false;
+
+    private static float InputCurve(float x)
```
Comment (NeerajW) [inline Assets/HoloToolkit/Utilities/Scripts/ManualCameraControl.cs]: Is this function trying to smooth mouse input?

File: Assets/HoloToolkit/Utilities/Scripts/ManualCameraControl.cs [hunk 6]
```diff
@@ -0,0 +1,313 @@
...
+    {
+        // smoothing input curve, converts from [-1,1] to [-2,2]
+        return Mathf.Sign(x) * (1.0f - Mathf.Cos(0.5f * Mathf.PI * Mathf.Clamp(x, -1.0f, 1.0f)));
+    }
+
+#if UNITY_EDITOR
```
Comment (NeerajW) [inline Assets/HoloToolkit/Utilities/Scripts/ManualCameraControl.cs]: Perhaps refactor this into a new class in the Utilities folder called DLLImports?

**Comment by jwittner** [comment]
Maybe the default control scheme could match the Emulator without dropping the ability to customize?

**Comment by jochao** [comment]
The script so far is just able to move and look, and you can map the keys to match the emulator. Emulator uses WASD to move and the left mouse button to look.  The script uses WASD and can be set in the editor to use left mouse to look.

**Comment by jwittner** [comment]
Sounds great! Looking forward to this functionality.

File: Assets/HoloToolkit/Utilities/Scripts/ManualCameraControl.cs [hunk 7]
```diff
@@ -0,0 +1,313 @@
...
+            return false;
+        }
+    }
+
+    private void OnApplicationFocus(bool focusStatus)
+    {
+        this.appHasFocus = focusStatus;
+    }
+
+    private void SetWantsMouseJumping(bool wantsJumping)
```
Comment (jochao) [inline Assets/HoloToolkit/Utilities/Scripts/ManualCameraControl.cs]: Should we simplify and remove the mouse jumping?

**Pull request merged**